vendor/github.com/OpenBazaar/multiwallet/service/wallet_service.go

```diff
@@ -125,7 +125,7 @@ func (ws *WalletService) listen() {
 		blockChan = ws.client.BlockNotify()
 	)
 	for _, sa := range addrs {
-		ws.client.ListenAddress(sa.Addr)
+		ws.client.ListenAddresses(sa.Addr)
```

Comment:
ws.client.ListenAddresses should not be called for every individual address but instead for the entire "addrs" slice. Here is an example of how I handled it but you do you.

```suggestion
		func (ws *WalletService) listen() {
	var (
		addrs     = ws.getStoredAddresses()
		txChan    = ws.client.TransactionNotify()
		blockChan = ws.client.BlockNotify()
		addrsList []btcutil.Address
	)

	for _, sa := range addrs {
		addrsList = append(addrsList, sa.Addr)
	}
	ws.client.ListenAddresses(addrsList...)
	for {
		select {
		case <-ws.doneChan:
			return
		case tx := <-txChan:
			go ws.ProcessIncomingTransaction(tx)
		case block := <-blockChan:
			go ws.processIncomingBlock(block)
		}
	}
}
```

Reply:
Yeah I think through several iterations of this it got lost on this one. I will update.